ignite/metrics/mean_average_precision.py

```diff
@@ -0,0 +1,133 @@
+__all__ = ["MeanAveragePrecision"]
+
+from collections import defaultdict
+from typing import Callable, List, Optional, Tuple, Union
+
+import torch
+
+from ignite.metrics.metric import Metric
+
+box_iou = None
+
+
+class MeanAveragePrecision(Metric):
+    def __init__(
+        self,
+        iou_thresholds: Optional[Union[List[float], torch.Tensor]] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = torch.device("cpu"),
+    ) -> None:
+        r"""Calculate the mean average precision of overall categories.
+
+        Args:
+            iou_thresholds: list of IoU thresholds to be considered for computing Mean Average Precision.
+            output_transform: a callable that is used to transform the
+                :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+                form expected by the metric. This can be useful if, for example, you have a multi-output model and
+                you want to compute the metric with respect to one of the outputs.
+                By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+            device: specifies which device updates are accumulated on. Setting the
+                metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+                non-blocking. By default, CPU.
+        """
+        global box_iou
+        try:
+            from torchvision.ops import box_iou
```

Comment:
Let's make it in the following way:
```suggestion     
            from torchvision.ops import box_iou
            self.box_iou = box_iou
```